src/pages/home/report/ReportActionCompose.js

```diff
@@ -323,6 +391,36 @@ class ReportActionCompose extends React.Component {
         this.setState({maxLines});
     }
 
+    /**
+     * Replace the code of emoji and update selection
+     * @param {Number} highlightedEmojiIndex
+     */
+    suggestedEmojisSelect(highlightedEmojiIndex) {
+        this.updateComment(this.state.value.slice(0, this.state.colonIndex)
+        + this.state.suggestedEmojis[highlightedEmojiIndex].code
+        + this.state.value.slice(this.state.colonIndex).replace(/^:\w+($|\s)/, ' '));
+
+        this.setState(prevState => ({
+            selection: {
+                start: prevState.colonIndex + 2,
+                end: prevState.colonIndex + 2,
+            },
+        }));
+    }
+
+    /**
+     * Check if this piece of string looks like an emoji
+     * @param {Number} str
+     * @param {Number} pos
+     * @returns {Boolean}
+     */
+    isLikeEmojiCode(str, pos) {
+        const newLineOrWhiteSpaces = /[\n\s]/g;
+        const leftWords = str.slice(0, pos).split(newLineOrWhiteSpaces);
+        const leftWord = leftWords[leftWords.length - 1];
```

Comment:
```suggestion
        const leftWord = _.last(leftWords);
```

Safer because we'll never try to access an array index out of bounds (such as `-1`). Also just a bit more readable.